### ISSUE TITLE
vcpkg master fails to build with VS2015 as found in AppVeyor image

### ISSUE THREAD

**Issue by rouault** [microsoft/vcpkg#14637]
**Describe the bug**

**Environment**
- OS: Windows
- Compiler: Visual Studio 2015 from "Visual Studio 2015" appveyor image

**To Reproduce**
bootstrap-vcpkg.bat

**Expected behavior**
It should compile

**Failure logs** 

Extract from https://ci.appveyor.com/project/OSGeo/gdal/builds/36387800
```
..\src\vcpkg.cpp(149): error C2280: 'vcpkg::Util::LockGuardPtr<T>::LockGuardPtr(const vcpkg::Util::LockGuardPtr<T> &)': attempting to reference a deleted function [C:\projects\gdal\vcpkg\toolsrc\windows-bootstrap\vcpkg.vcxproj]
          with
          [
              T=vcpkg::Metrics::Metrics
          ]
  C:\projects\gdal\vcpkg\toolsrc\include\vcpkg/base/lockguarded.h(34): note: compiler has generated 'vcpkg::Util::LockGuardPtr<T>::LockGuardPtr' here
          with
          [
              T=vcpkg::Metrics::Metrics
          ]
C:\projects\gdal\vcpkg\scripts\bootstrap.ps1 : Building vcpkg.exe failed.
```

**Comment by NancyLi1013** [comment]
Hi @rouault 

Thanks for posting this issue.

@strega-nil 

Could you please help take a look?

I can reproduce this on my machine with Visual Studio 2015 too.

```
PS E:\Nancy\test\vcpkg> .\bootstrap-vcpkg.bat

Building vcpkg.exe ...

  pch.cpp
  archives.cpp
  checks.cpp
  chrono.cpp
  cofffilereader.cpp
  downloads.cpp
  enums.cpp
  files.cpp
  hash.cpp
  json.cpp
  machinetype.cpp
  parse.cpp
  strings.cpp
  stringview.cpp
  system.cpp
  system.print.cpp
  system.process.cpp
  unicode.cpp
  binarycaching.cpp
  binaryparagraph.cpp
  build.cpp
  buildenvironment.cpp
  cmakevars.cpp
  commands.autocomplete.cpp
  commands.buildexternal.cpp
  commands.cache.cpp
  commands.ci.cpp
  commands.ciclean.cpp
..\src\vcpkg\binarycaching.cpp(1125): error C2280: 'vcpkg::Util::LockGuardPtr<T>::LockGuardPtr(const vcpkg::Util::LockG
uardPtr<T> &)': attempting to reference a deleted function [E:\Nancy\test\vcpkg\toolsrc\windows-bootstrap\vcpkg.vcxproj
]
          with
          [
              T=vcpkg::Metrics::Metrics
          ]
  E:\Nancy\test\vcpkg\toolsrc\include\vcpkg/base/lockguarded.h(34): note: compiler has generated 'vcpkg::Util::LockGuar
  dPtr<T>::LockGuardPtr' here
          with
          [
              T=vcpkg::Metrics::Metrics
          ]
  commands.contact.cpp
..\src\vcpkg\build.cpp(793): error C2280: 'vcpkg::Util::LockGuardPtr<T>::LockGuardPtr(const vcpkg::Util::LockGuardPtr<T
> &)': attempting to reference a deleted function [E:\Nancy\test\vcpkg\toolsrc\windows-bootstrap\vcpkg.vcxproj]
          with
          [
              T=vcpkg::Metrics::Metrics
          ]
  E:\Nancy\test\vcpkg\toolsrc\include\vcpkg/base/lockguarded.h(34): note: compiler has generated 'vcpkg::Util::LockGuar
  dPtr<T>::LockGuardPtr' here
          with
          [
              T=vcpkg::Metrics::Metrics
          ]
  commands.cpp
  commands.create.cpp
  commands.dependinfo.cpp
  commands.edit.cpp
  commands.env.cpp
  commands.fetch.cpp
  commands.format-manifest.cpp
  commands.hash.cpp
  commands.info.cpp
  commands.integrate.cpp
  commands.list.cpp
..\src\vcpkg\commands.integrate.cpp(431): error C2280: 'vcpkg::Util::LockGuardPtr<T>::LockGuardPtr(const vcpkg::Util::L
ockGuardPtr<T> &)': attempting to reference a deleted function [E:\Nancy\test\vcpkg\toolsrc\windows-bootstrap\vcpkg.vcx
proj]
          with
          [
              T=vcpkg::Metrics::Metrics
          ]
  E:\Nancy\test\vcpkg\toolsrc\include\vcpkg/base/lockguarded.h(34): note: compiler has generated 'vcpkg::Util::LockGuar
  dPtr<T>::LockGuardPtr' here
          with
          [
              T=vcpkg::Metrics::Metrics
          ]
  commands.owns.cpp
  commands.porthistory.cpp
  commands.portsdiff.cpp
  commands.search.cpp
  commands.setinstalled.cpp
  commands.upgrade.cpp
  commands.upload-metrics.cpp
  commands.version.cpp
  commands.xvsinstances.cpp
  configuration.cpp
  dependencies.cpp
  export.cpp
  export.chocolatey.cpp
  export.ifw.cpp
  export.prefab.cpp
  globalstate.cpp
  help.cpp
  input.cpp
  install.cpp
  platform-expression.cpp
  metrics.cpp
  packagespec.cpp
  paragraphs.cpp
  portfileprovider.cpp
  postbuildlint.buildtype.cpp
  postbuildlint.cpp
  registries.cpp
  remove.cpp
  sourceparagraph.cpp
  statusparagraph.cpp
  statusparagraphs.cpp
  tools.cpp
  triplet.cpp
  update.cpp
  userconfig.cpp
  vcpkgcmdarguments.cpp
  vcpkglib.cpp
  vcpkgpaths.cpp
  versiont.cpp
  visualstudio.cpp
  vcpkg.cpp
..\src\vcpkg.cpp(149): error C2280: 'vcpkg::Util::LockGuardPtr<T>::LockGuardPtr(const vcpkg::Util::LockGuardPtr<T> &)':
 attempting to reference a deleted function [E:\Nancy\test\vcpkg\toolsrc\windows-bootstrap\vcpkg.vcxproj]
          with
          [
              T=vcpkg::Metrics::Metrics
          ]
  E:\Nancy\test\vcpkg\toolsrc\include\vcpkg/base/lockguarded.h(34): note: compiler has generated 'vcpkg::Util::LockGuar
  dPtr<T>::LockGuardPtr' here
          with
          [
              T=vcpkg::Metrics::Metrics
          ]
E:\Nancy\test\vcpkg\scripts\bootstrap.ps1 : Building vcpkg.exe failed. Please ensure you have installed Visual Studio
with the Desktop C++ workload and the Windows SDK for Desktop C++.
At line:1 char:4
+ & {& 'E:\Nancy\test\vcpkg\scripts\bootstrap.ps1' }
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,bootstrap.ps1

ScriptHalted
At E:\Nancy\test\vcpkg\scripts\bootstrap.ps1:411 char:5
+     throw
+     ~~~~~
    + CategoryInfo          : OperationStopped: (:) [], RuntimeException
    + FullyQualifiedErrorId : ScriptHalted
```

**Comment by strega-nil** [comment]
Fixed by #14659

**Comment by NancyLi1013** [comment]
Hi @rouault 

This issue should be fixed now. Please use `git pull `to update vcpkg to the latest commit and then rebuild this.
